tests/test_wfp/test_wfp.py

```diff
@@ -0,0 +1,1451 @@
+    """
+    country_code = 1
+    indicator_code = 100
+
+    # Mock response with error
+    mock_response_inflation.side_effect = requests.exceptions.RequestException(
+        "Mocked request error"
+    )
+
+    # Assert that DataExtrationError is raised when `extract_data` method is called
+    with pytest.raises(
+        DataExtractionError, match="Error getting inflation data: Mocked request error"
+    ):
+        wfp_inflation.extract_data(country_code, indicator_code)
+
+
+def test_wfp_inflation_format_data_successful(wfp_inflation):
+    """
+    Test that WFPInflation's `format_data` method processes valid data into the expected DataFrame.
+    """
+    # Mock data
+    test_csv = """IndicatorName,CountryName,Date,Value,SourceOfTheData
+    Headline inflation (YoY),Ecuador,31/10/2024,5.1,MockSource"""
+    data = io.BytesIO(test_csv.encode("utf-8"))
+    indicator_name = "Headline inflation (YoY)"
+    iso3_code = "ECU"
+
+    # Create expected DataFrame
+    expected_df = pd.DataFrame(
+        {
+            "date": pd.to_datetime(["2024-10-31"]),
+            "value": [5.1],
+            "source": ["MockSource"],
+            "indicator_name": ["Headline inflation (YoY)"],
+            "iso3_code": ["ECU"],
+            "country_name": ["Ecuador"],
+            "unit": ["percent"],
+        }
+    ).pipe(convert_dtypes)
+
+    # Call `format_data` method
+    result_df = wfp_inflation.format_data(data, indicator_name, iso3_code)
+
+    # Assert that resulted and expected DataFrames are equal
+    pd.testing.assert_frame_equal(result_df, expected_df)
+
+
+def test_wfp_inflation_format_data_exception(wfp_inflation):
+    """
+    Test that WFPInflation's `format_data` method raises DataFormattingError for invalid data.
+    """
+    # Mock data
+    invalid_data = io.BytesIO(b"Invalid,CSV,Data")
+    indicator_name = "Inflation"
+    iso3_code = "TST"
+
+    # Assert that DataFormattingError is raised when `format_data` method is called
+    with pytest.raises(
+        DataFormattingError, match="Error formatting data for country - TST:"
+    ):
+        wfp_inflation.format_data(invalid_data, indicator_name, iso3_code)
+
+
+def test_wfp_inflation_load_data_all_countries_loaded(wfp_inflation_pre_load):
+    """
+    Test that WFPInflation's `load_data` method does nothing and logs nothing when all countries are already loaded.
+    """
+    # Add data to class
+    wfp_inflation_pre_load._data["Headline inflation (YoY)"] = {
+        "USA": mock.Mock(),
+        "CAN": mock.Mock(),
+    }
+
+    # Mock logs
+    with mock.patch("bblocks_data_importers.config.logger.info") as mock_logger:
+        # Call `load_data` method
+        wfp_inflation_pre_load.load_data("Headline inflation (YoY)", ["USA", "CAN"])
+        # Assert that no messages are logged
+        mock_logger.assert_not_called()
+
+
+def test_wfp_inflation_load_data_country_not_available(wfp_inflation_pre_load):
```

Comment:
You should also test `_load_data` when some data is already loaded and some is not